src/main/java/uk/org/tombolo/core/utils/SubjectUtils.java

```diff
@@ -135,11 +136,13 @@ private static Query queryFromSubjectSpecification(Session session, SubjectRecip
 
 		// Add Geo Match Rule if exists
 		if (null != subjectRecipe.getGeoMatchRule()){
-			if (subjectRecipe.getGeoMatchRule().geoRelation == SubjectRecipe.SubjectGeoMatchRule.GeoRelation.within){
-				hqlQuery += " and within(shape, :geom) = true";
+			List<SubjectRecipe.SubjectGeoMatchRule.GeoRelation> geoRel = new ArrayList<>();
+			Collections.addAll(geoRel, SubjectRecipe.SubjectGeoMatchRule.GeoRelation.values());
+			if (geoRel.contains(subjectRecipe.getGeoMatchRule().geoRelation)){
+				hqlQuery += " and " + subjectRecipe.getGeoMatchRule().geoRelation.name() + "(shape, :geom) = true";
 			}else{
 				throw new IllegalArgumentException(String.format(
-						"SubjectGeoMatchRule attribute is not a valid type (is %s, can only be within)",
+						"SubjectGeoMatchRule attribute is not a valid type: ",
```

Comment:
Is there any way that we can make this exception print a list of the acceptable types?

Reply:
Yes, there is and I will add it.

Reply:
Cool - let me know once this is implemented and will approve. Would be good if @thanosbnt could look at it too.

Reply:
I think it works great. For a future test implementation and for the sake of completeness i think we could include testing all geometries against all the others (where it makes sense).